### PR TITLE
Release v3.20.4

### PR DESCRIPTION
**Pull request by monde**
* CHANGELOG entry
* client's user-agent updated
* small commit 9f4f4f8c180cbf24100debe0a4f7e2e8b26e096c from a @ymylei PR
* code clean up suggestions from `make vet`

### CODE REVIEW

**Comment by virgofx** [comment]
Looks like Enhancements got bumped out of section

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,26 @@
+ENHANCEMENTS:
+
```
Comment (virgofx) [inline CHANGELOG.md]: Looks like `ENHANCEMENTS:` is bumped out of it's respective section. I think it just needs to be moved down and should be good.

**Review by virgofx — APPROVED** [review]
🎉 I notice the other `SpCertificate` commit injected in release PR. It's tiny and small so not a huge issue

Edit: I think one more edit needs to be done for the user agent also: [okta/config.goL#104](https://github.com/okta/terraform-provider-okta/blob/master/okta/config.go#L104)

File: CHANGELOG.md (hunk 1 above)
Comment (monde) [reply CHANGELOG.md]: 👍

**Comment by monde** [comment]
> 🎉 I notice the other `SpCertificate` commit injected in release PR. It's tiny and small so not a huge issue
> 
> Edit: I think one more edit needs to be done for the user agent also: [okta/config.goL#104](https://github.com/okta/terraform-provider-okta/blob/master/okta/config.go#L104)

I cherry picked in that commit as I didn't want the go module changes with the PR it was in. Good eyes on the agent version!

**Review by vijetmahabaleshwar-okta — APPROVED** [review]

**Comment by virgofx** [comment]
Yay -- but do my eyes deceive me - I didn't see user agent config change 🤷

**Comment by monde** [comment]
bonk, didn't save the user agent change, will update in a v3.20.5 release.

**Comment by virgofx** [comment]
Haha. It happens. I noticed Terraform Registry didn't update and thus can't pull 3.20.5 :( Not sure if the goreleaser Actions didn't work or whatnot.

**Pull request merged**